### DISCUSSION TITLE
Click Handling

### DISCUSSION THREAD

**Discussion by EMIN3M1986** [patrykandpatrick/vico#506] [Questions]
How to detect user click chart? 
For example user click on a bar of Barchart. I want to have index of the bar or the data of the bar. 
How can I do that?

**Comment by Gowsky** [comment]
Hi @EMIN3M1986, [`MarkerVisibilityChangeListener#onMarkerShown`](https://patrykandpatrick.com/vico/api/2.0.0-alpha.5/core/com.patrykandpatrick.vico.core.marker/-marker-visibility-change-listener/?query=interface%20MarkerVisibilityChangeListener#-849483284%2FFunctions%2F186187275) is invoked when a user taps on a chart, and it yields marked entries.
For Jetpack Compose, you can pass the listener as an argument of `CartesianChartHost` function, and for views you can set it with `CartesianChartView#markerVisibilityChangeListener`.

**Reply by patrickmichalik** [reply]
Hello! I wanted to expand on this and mention that if you need the functionality of `MarkerVisibilityChangeListener` but don’t want your chart to have a `Marker`, you can use a custom `Marker` implementation that doesn’t draw anything or request any insets. We’ll be looking into separating `Marker`s from selection detection.

**Reply by Akash-pugazh** [reply]
can i get a example code for that :) because it is a argument markerVisibilityChangeListener that also accepts MarkerVisibilityChangeListener type of not sure how to do it, 
My scenario is to i have a series of points whenever a point is clicked on the graph i need to set it as active (persistent marker state) like, Please provide me a solution for this case. Thanks in advance ❣️

**Reply by Gowsky** [reply]
Hi @Akash-pugazh! [`MarkerVisibilityChangeListener`](https://patrykandpatrick.com/vico/api/2.0.0-alpha.12/core/com.patrykandpatrick.vico.core.marker/-marker-visibility-change-listener/?query=interface%20MarkerVisibilityChangeListener) is an interface you need to implement just like any other Kotlin interface. Inside of overridden functions you can [`set`](https://patrykandpatrick.com/vico/api/2.0.0-alpha.12/core/com.patrykandpatrick.vico.core.chart/-cartesian-chart/#-1870900852%2FFunctions%2F186187275), [`add`](https://patrykandpatrick.com/vico/api/2.0.0-alpha.12/core/com.patrykandpatrick.vico.core.chart/-cartesian-chart/#911991383%2FFunctions%2F186187275), or [`remove`](https://patrykandpatrick.com/vico/api/2.0.0-alpha.12/core/com.patrykandpatrick.vico.core.chart/-cartesian-chart/#-1713665082%2FFunctions%2F186187275) persistent markers.
